docs/source/howto/run_codes.rst

```diff
@@ -357,24 +357,215 @@ After :ref:`setting up your computer <how-to:run-codes:computer>` and :ref:`sett
+
+.. code-block:: yaml
+
+    profile-name:
+        default: False
+        enabled:
+            - aiida.calculations:quantumespresso.*
+            - '*.seekpath_structure_analysis'
+
+You can even override a wildcard with a more specific entry.
+The following configuration enables caching for all ``aiida.calculation`` entry points, except those of ``aiida-quantumespresso``:
+
+.. code-block:: yaml
+
+    profile-name:
+        default: False
+        enabled:
+            - aiida.calculations:*
+        disabled:
+            - aiida.calculations:quantumespresso.*
+
+
+Instance level
+..............
+
+Even when caching is turned off for a given calculation type, you can enable it on a case-by-case basis by using the :class:`~aiida.manage.caching.enable_caching` context manager for testing purposes:
+
+.. code-block:: python
+
+    from aiida.engine import run
+    from aiida.manage.caching import enable_caching
+    with enable_caching(identifier='aiida.calculations:templatereplacer'):
+       run(...)
+
+.. warning::
+
+    This affects only the current python interpreter and won't change the behavior of the daemon workers.
+    This means that this technique is only useful when using :py:class:`~aiida.engine.run`, and **not** with :py:class:`~aiida.engine.submit`.
+
+If you suspect a node is being reused in error (e.g. during development), you can also manually *prevent* a specific node from being reused:
+
+1. Load one of the nodes you suspect to be a clone.
+   Check that :meth:`~aiida.orm.nodes.Node.get_cache_source` returns a UUID.
+   If it returns `None`, the node was not cloned.
+2. Clear the hashes of all nodes that are considered identical to this node:
+
+    .. code:: python
+
+        for node in node.get_all_same_nodes():
+            node.clear_hash()
+3. Run your calculation again.
+   The node in question should no longer be reused.
+
+
+.. _how-to:run-codes:caching:limitations:
+
+Limitations
+-----------
+
+#. Workflow nodes are not cached.
+   In the current design this follows from the requirement that the provenance graph be independent of whether caching is enabled or not:
+
+   * **Calculation nodes:** Calculation nodes can have data inputs and create new data nodes as outputs.
+     In order to make it look as if a cloned calculation produced its own outputs, the output nodes are copied and linked as well.
+   * **Workflow nodes:** Workflows differ from calculations in that they can *return* an input node or an output node created by a calculation.
+     Since caching does not care about the *identity* of input nodes but only their *content*, it is not straightforward to figure out which node to return in a cached workflow.
+
+   For the moment, this limitation is acceptable since the runtime of AiiDA WorkChains is usually dominated by expensive calculations, which are covered by the current caching mechanism.
```

Comment:
```suggestion
   This limitation has typically no significant impact since the runtime of AiiDA WorkChains is commonly dominated by expensive calculations, which are covered by the current caching mechanism.
```
Let's state this more factual and leave it up to the user/reader to decide whether that is acceptable.